src/Apps/Order/Routes/Payment/types.ts

```diff
@@ -0,0 +1,26 @@
+export type PaymentState = {
+  selectedBankAccountId: string
+}
+
+export enum PaymentActions {
+  SET_SELECTED_BANK_ACCOUNT_ID = "SET_SELECTED_BANK_ACCOUNT_ID",
+}
+
+export type PaymentAction = ActionMap<PaymentActionsPayload>[keyof ActionMap<
+  PaymentActionsPayload
+>]
+
+type PaymentActionsPayload = {
+  [PaymentActions.SET_SELECTED_BANK_ACCOUNT_ID]: string
+}
+
+type ActionMap<M extends { [index: string]: any }> = {
+  [Key in keyof M]: M[Key] extends undefined
+    ? {
+        type: Key
+      }
+    : {
+        type: Key
+        payload: M[Key]
+      }
+}
```

Comment:
totally fine to keep a types file! I have found it easier in experience to keep types related to reducer / context all inside the context file tho just for ease of access. But no strong opinions here.

Reply:
I think it makes sense, will move!